src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,114 @@
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the profile page. */
+public class ProfileServlet extends HttpServlet {
+
+  /** Store class that gives access to Users. */
+  private UserStore userStore;
+
+  /** Store class that gives access to Profiles. */
+  private ProfileStore profileStore;
+
+  /** Store class that gives access to Profiles. */
+  private MessageStore messageStore;
+
+  /**
+   * Set up state for handling conversation-related requests. This method is only called when
+   * running in a server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setProfileStore(ProfileStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  /**
+   * Sets the ProfileStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setProfileStore(ProfileStore userStore) {
+    this.profileStore = profileStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method
+   * for use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * This function fires when a user navigates to the profile page. It gets all of the
+   * profiles and messages from the model and forwards to profiles.jsp for rendering the list.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+    Profile profile = profileStore.getProfile();
+    List<Message> messages = messageStore.getAllMessages();
+    request.setAttribute("profile", profile);
+    request.setAttribute("messages", setMessages(messages));
+    request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
+  }
+
+  /**
+   * This function fires when a user edits the description on the profile page. 
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+        Profile profile = profileStore.getProfile(userID);
+
+
+    String description = request.getParameter("description");
+
+    Profile description =
```

Comment:
Instead of creating a new profile, you want to use the updateProfile() method from ProfileStore because we are just updating the description on the existing profile.